### PR TITLE
Fix archival of polymorphic association of another model with same id

### PR DESCRIPTION
**Pull request by gsmetal**
If we have two models (with different tables) holding same has_many polymorphic association than archiving of one model object will archive associated models of another model object with same id. Association scope building was not taking type into account, relying only on id. This PR fixes that.

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -79,9 +80,9 @@ def require_test_classes
   end
 
   fixtures = if ActiveRecord::VERSION::MAJOR >= 4
-               RAILS_5_FIXTURE_CLASSES + BASE_FIXTURE_CLASSES
+               RAILS_5_FIXTURE_CLASSES
```
Comment (janxious) [inline test/test_helper.rb]: what's with this change?

File: gemfiles/rails_4.1.gemfile [hunk 2]
```diff
@@ -4,4 +4,4 @@ source "https://rubygems.org"
 
 gem "rails", "~> 4.1.0"
 
-gemspec :path => "../"
+gemspec path: "../"
```
Comment (janxious) [inline gemfiles/rails_4.1.gemfile]: these files are generated by appraisal, so did you get a change here or did you make this change?

File: test/test_helper.rb (hunk 1 above)
Comment (gsmetal) [reply test/test_helper.rb]: `BASE_FIXTURE_CLASSES` are added just after `if` block, I removed duplication. This change is not necessity for PR, just a little code cleaning. Should I revert it?

Comment (janxious) [reply test/test_helper.rb]: Nah, I just didn't notice the duplication. 👍

File: gemfiles/rails_4.1.gemfile (hunk 2 above)
Comment (janxious) [reply gemfiles/rails_4.1.gemfile]: @gsmetal could you tell me about this one?

Comment (gsmetal) [reply gemfiles/rails_4.1.gemfile]: Oh.. I wrote about this and leave it as `Pending` review change 😕

I executed `appraisal install` with newer version of appraisal so it made this change. I tried to add rails 5.2 Appraisal but it had problems with `"table_name is set to 'legacy'"` test (with parentheses in SQL) so I didn't dig into that and removed it.

Comment (gsmetal) [reply gemfiles/rails_4.1.gemfile]: BTW, there are test problems on travis with older versions of rails and ruby caused by bundle 2.0 release.

Comment (janxious) [reply gemfiles/rails_4.1.gemfile]: I don't suppose you have a link about mitigating those failures? I assume I can just tell travis to stick to old version or something, or there's a script, but I'm porbably being naively hopeful.

Comment (gsmetal) [reply gemfiles/rails_4.1.gemfile]: No, I don't know quick way to fix all of this new dependencies between new RVM, RubyGems 3.0, Bundler 2.0 and older versions of Ruby and Rails in Travis.

Comment (gsmetal) [reply gemfiles/rails_4.1.gemfile]: I assume you should pin rubygems and bundler version below 3.0 and 2.0 respectively here https://github.com/expectedbehavior/acts_as_archival/blob/87fe14e610bd42f7caa679613b59b01eb8efe620/.travis.yml#L20-L21

Comment (janxious) [reply gemfiles/rails_4.1.gemfile]: I dug in on this one on #39 and fixed it there. I'll check on these changes again and merge them if I still agree with them :)

**Comment by janxious** [comment]
Thanks for the patch!

**Pull request merged**